Cache: Error: mkdir(/Users/runner/.cache/dune/db/temp): File exists

https://github.com/ocsigen/lwt/runs/5262157265?check_suite_focus=true

```
+ /Users/runner/.opam/opam-init/hooks/sandbox.sh "build" "dune" "build" "-p" "csexp" "-j" "3" "@install" (CWD=/Users/runner/work/lwt/lwt/_opam/.opam-switch/build/csexp.1.5.1)
[ERROR] The compilation of csexp.1.5.1 failed at "dune build -p csexp -j 3 @install".
- Error: mkdir(/Users/runner/.cache/dune/db/temp): File exists
- Error: mkdir(/Users/runner/.cache/dune/db/temp): File exists
```

--- a/otherlibs/stdune/fpath.ml
+++ b/otherlibs/stdune/fpath.ml
@@ -19,8 +19,8 @@ type mkdir_p_result =
   | Already_exists
   | Created
 
-let rec mkdir_p ?(perms = 0o777) t_s =
-  match mkdir ~perms t_s with
+let rec mkdir_p ?perms t_s =
+  match mkdir ?perms t_s with
   | Created -> Created
   | Already_exists -> Already_exists
   | Missing_parent_directory -> (
@@ -31,8 +31,8 @@ let rec mkdir_p ?(perms = 0o777) t_s =
         []
     else
       let parent = Filename.dirname t_s in
-      match mkdir_p ~perms parent with
-      | Created | Already_exists ->
+      match mkdir_p ?perms parent with
+      | Created | Already_exists -> (
         (* The [Already_exists] case might happen if some other process managed
            to create the parent directory concurrently. *)
         Unix.mkdir t_s perms;

